apiserver/authentication/user.go

```diff
@@ -133,7 +133,7 @@ func (u *LocalUserAuthenticator) Authenticate(
 	// We believe we've got a password, so we'll try to authenticate with it.
 	// This will check the user service for the user, ensuring that the user
 	// isn't disabled or deleted.
-	user, err := u.UserService.GetUserByAuth(ctx, userTag.Name(), auth.NewPassword(authParams.Credentials))
+	user, err := u.UserService.GetUserByAuth(ctx, userTag.Id(), auth.NewPassword(authParams.Credentials))
```

Comment:
I don't mind being consistent, but technically, this one does not need to change, because and external user will not have a password and not be able to log in via that method.